### PR TITLE
[ISSUE #849]Solving the performance problem of MQClientInstance

### PR DESCRIPTION
**Pull request by ymwneu**
## What is the purpose of the change

Solving [issue](https://github.com/apache/rocketmq/issues/849)

## Brief changelog

MQClientInstance.java
add a cache for method findBrokerVersion.

### CODE REVIEW

**Comment by dongeforever** [comment]
@duhengforever Please help a review

File: client/src/main/java/org/apache/rocketmq/client/impl/factory/MQClientInstance.java [hunk 1]
```diff
@@ -805,6 +805,18 @@ private boolean isNeedUpdateTopicRouteInfo(final String topic) {
         return result;
     }
 
+    private void putVersionInfo(String brokerName, String brokerAddr, int version) {
+
+        if (!this.brokerVersionTable.containsKey(brokerName)) {
+            this.brokerVersionTable.put(brokerName, new HashMap<String, Integer>(4));
```
Comment (duhenglucky) [inline client/src/main/java/org/apache/rocketmq/client/impl/factory/MQClientInstance.java]: Why init this map size to 4？

File: client/src/main/java/org/apache/rocketmq/client/impl/factory/MQClientInstance.java [hunk 2]
```diff
@@ -102,6 +102,8 @@
         new ConcurrentHashMap<String, HashMap<Long, String>>();
     private final ConcurrentMap<String/* Broker Name */, HashMap<String/* address */, Integer>> brokerVersionTable =
         new ConcurrentHashMap<String, HashMap<String, Integer>>();
+    private final ConcurrentMap<String/* Broker Name */, HashMap<String/* address */, Long>> versionUpdateTimeTable =
```
Comment (duhenglucky) [inline client/src/main/java/org/apache/rocketmq/client/impl/factory/MQClientInstance.java]: It seems that the versionUpdateTimeTable and  brokerVersionTable  would be difficult to keep insistence.

File: client/src/main/java/org/apache/rocketmq/client/impl/factory/MQClientInstance.java [hunk 3]
```diff
@@ -1042,9 +1054,35 @@ public FindBrokerResult findBrokerAddressInSubscribe(
     }
 
     public int findBrokerVersion(String brokerName, String brokerAddr) {
```
Comment (duhenglucky) [inline client/src/main/java/org/apache/rocketmq/client/impl/factory/MQClientInstance.java]: Why not record the broker version after sendbeatheart？

**Comment by ShannonDing** [comment]
@ymwneu this pr will be closed as many conflicts and time too long, welcome to create a new PR to trace this issue.

**Pull request closed without merging**